unixHttpRoutes.go

```diff
@@ -276,6 +280,86 @@ func handleDeleteLogFilter(w http.ResponseWriter, r *http.Request, _ httprouter.
 	w.WriteHeader(http.StatusOK)
 }
 
+func handleGetDNS(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	inst := networking.GetInstance()
+	if inst != nil {
+		dnsarray, err := inst.GetDNS()
+		if err == nil {
+			w.WriteHeader(http.StatusOK)
+			dnsjson, _ := json.Marshal(dnsarray)
+			w.Write(dnsjson)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(fmt.Sprintf("{\"error\":\"%s\"}", err.Error())))
+		}
+	} else {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("{\"error\":\"no network manager\"}"))
+	}
+	defer r.Body.Close()
+
+}
+
+func handlePostDNS(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	inst := networking.GetInstance()
+
+	config := new(maestroSpecs.NetworkConfigPayload)
+
+	if inst != nil {
+		body, err := ioutil.ReadAll(r.Body)
+		if err == nil {
+			json.Unmarshal(body, &config)
+			dns := config.Nameservers[0]
```

Comment:
should this be unrolled to AddDNS() on each Nameserver in the array?

Reply:
done